pkg/controller/user-cluster-controller-manager/constraint-template-controller/controller_test.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2020 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package constrainttemplatecontroller
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	"github.com/open-policy-agent/frameworks/constraint/pkg/apis/templates/v1beta1"
+
+	v1 "k8c.io/kubermatic/v2/pkg/crd/kubermatic/v1"
+	kubermaticlog "k8c.io/kubermatic/v2/pkg/log"
+
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/diff"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/record"
+	fakectrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func TestReconcile(t *testing.T) {
+	testCases := []struct {
+		name               string
+		constraintTemplate *v1.ConstraintTemplate
+		requestName        string
+		expectedCT         *v1.ConstraintTemplate
+	}{
+		{
+			name:               "sync ct to user cluster",
+			constraintTemplate: genConstraintTemplate("requiredlabels"),
+			requestName:        "requiredlabels",
+			expectedCT:         genConstraintTemplate("requiredlabels"),
+		},
+	}
+
+	for idx := range testCases {
```

Comment:
You can use this loop instead, and then you don't need to worry about the index in the next line:
```go
for _, tc := range testCases {
```
so you can remove line 53